### ISSUE TITLE
[BUG] Inconsistent behavior when launching app from Docker

### ISSUE THREAD

**Issue by agirbau** [voxel51/fiftyone#4589]
### Describe the problem

TL; DR: When launching a session `session = fo.launch_app(dataset, address="localhost", auto=False)` within docker sometimes works to suddenly stop working randomly.

I'm running fiftyone 0.24.1 from a docker container locally from my computer (macbook M3) from a jupyter notebook. It works as expected but, after modifying some datasets by using an external script, it randomly stops working after refreshing the fiftyone app. Randomly means that sometimes it refreshes correctly the data, and sometimes it just shows "this site can't be reached" and can't relaunch the app properly.

As extra information, `session = fo.launch_app()` returns "Session launched. Run `session.show()` to open the App in a cell output."

I have been testing around. The behavior I have seen so far is:
- Resetting the jupyter notebook doesn't work for both `0.0.0.0` and `localhost` won't work.
- Trying to run `session = fo.launch_app()` externally doesn't work.
- When resetting the docker container sometimes works but usually won't.
- When rebuilding the docker image usually works but sometimes won't.
- Resetting the PC sometimes works but usually won't.

My wild guess is that there is some kind of cache that is not freed or there is a problem with sockets being used. Apologizes for the lack of understanding in this area.

### System information

-   **OS Platform and Distribution** (e.g., Linux Ubuntu 22.04): iOS, Macbook Pro M3
-   **Python version** (`python --version`): 3.10
-   **FiftyOne version** (`fiftyone --version`): 0.24.1
-   **FiftyOne installed from** (pip or source): Docker pull

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another
member of your organization be willing to contribute a fix for this bug to the
FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently
-   [ ] Yes. I would be willing to contribute a fix for this bug with guidance
        from the FiftyOne community
-   [x] No. I cannot contribute a bug fix at this time (don't have the knowledge)

**Comment by agirbau** [comment]
I've run extra tests, check which files change and when, and realized that it seems to be a mongodb problem when updating a big dataset many times with:

`dataset.add_sample(sample)`

Basically it seems that mongodb internals go crazy and keep updating/resetting files, e.g. /fiftyone/db/collection-9--7546034396399586667.wt keeps updating every 10 seconds.

If instead I generate a list of samples and add them in one go with 

`dataset.add_samples(sample_list)`

seems that everything is OK for now. 

I understand that this can happen in a session, but I still don't understand why after resetting the docker image mongodb is not cleared with the reset...

**Comment by agirbau** [comment]
Final exploration and closing.

The behavior is consistent now, it runs smoothly and doesn't crash even if I modify the dataset.

I realized I was parsing the wrong arguments to `dataset.add_sample` (I was parsing a list of 3 samples instead of the appropriate arguments), which made mongodb (or something related) go crazy. 

Two final thoughts: 
1. I am not sure why these files were not deleted after resetting the docker environment, maybe it had something to do with the dataset being "persistent"?
2. I think both `datset.add_sample` and `datset.add_samples` should throw an exception instead of accepting random parameters to avoid possible conflicts with mongodb (or whatever is causing this inconsistency issue).

Closing the issue but it would be nice if someone could take a look.
